bcda/logging/middleware.go

```diff
@@ -90,3 +92,12 @@ func (l *StructuredLoggerEntry) Panic(v interface{}, stack []byte) {
 		"panic": fmt.Sprintf("%+v", v),
 	})
 }
+
+func Redact(uri string) string {
+	re := regexp.MustCompile(`Bearer%20([^&]+)(?:&|$)`)
+	submatches := re.FindAllStringSubmatch(uri, -1)
```

Comment:
is there a chance for this to return nil?

Reply:
There shouldn't be. If there are no matches, `FindAllStringSubmatch()` returns an empty slice, causing `Redact()` to return the original string. The third part of `TestRedact()` has an example of that, using a URL that doesn't have `Bearer` in it.